substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/tracing/TraceFileWriter.java

```diff
@@ -100,7 +101,7 @@ private static void printValue(JsonWriter json, Object value) throws IOException
         if (value instanceof byte[]) {
             s = Base64.getEncoder().encodeToString((byte[]) value);
         } else if (value != null) {
-            s = value.toString();
+            s = Objects.toString(value);
```

Comment:
I don't quite understand what the problem is. Before I recently added the `value != null` check above, `NullPointerException` could be thrown here, but now `null` will be passed to `JsonWriter.quote` which prints a JSON `null`. `Objects.toString` would return the string `"null"`, which is worse because it might be a legitimate value.

Reply:
Originally I observed this on a branch that was based on commit d8d914648604df9fa83c13aab35c81a37b61c027, and after rebasing to master I somehow failed to notice that this case is now handled. Thank you for a quick response!